opal/corelib/class.rb

```diff
@@ -59,7 +68,15 @@ def inherited(cls)
 
   def new(*args, &block)
     %x{
-      var object = #{allocate};
+      var object;
+
+      if (self.$$bridge && self.$allocate == Opal.Class.prototype.$allocate) {
```

Comment:
What if... we subclass a bridged class?

```ruby
class Array < `Array`; end
class MyArray < Array; end
class YourArray < MyArray; end
```

(The lack of awareness of that edge case caused in the past a hard-to-debug issue where we missed backtraces in Firefox - due to not launching a constructor - if Exception was subclassed more than once - see #2540 ).

Reply:
For comparison, we have something that's called `pristine` methods. This is an option for us to define a method to be unmodified from the default implementation. Perhaps this is a better idea.

Reply:
Alright, i will pay attention to that case, i still need to understand better whats going on